apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -99,9 +98,9 @@ where
             body.operation_name.to_owned(),
             request.query_plan_options,
         );
-        let cm = self.cm.clone();
+        let qp = self.clone();
```

Comment:
This clone is cheap right now due to use of `Arc` on the members of `CachingQueryPlanner`. We should have a comment at line 25 to note this is required to warn future engineers to be aware of why we use `Arc`.

Reply:
we rely on cloning an Arc in a lot of parts of the router, do we need to point it out every time?